packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -94,44 +105,73 @@ function wrapFunctions(userCode: string, templateCode: string, filepath: string)
  * Wrap `getStaticPaths`, `getStaticProps`, and `getServerSideProps` (if they exist) in the given page code
  */
 export default function wrapDataFetchersLoader(this: LoaderThis<LoaderOptions>, userCode: string): string {
-  // We know one or the other will be defined, depending on the version of webpack being used
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const { projectDir } = this.getOptions ? this.getOptions() : this.query!;
-
   // For now this loader only works for ESM code
   if (!isESM(userCode)) {
     return userCode;
   }
 
-  // If none of the functions we want to wrap appears in the page's code, there's nothing to do. (Note: We do this as a
-  // simple substring match (rather than waiting until we've parsed the code) because it's meant to be an
-  // as-fast-as-possible fail-fast. It's possible for user code to pass this check, even if it contains none of the
-  // functions in question, just by virtue of the correct string having been found, be it in a comment, as part of a
-  // longer variable name, etc. That said, when we actually do the code manipulation we'll be working on the code's AST,
-  // meaning we'll be able to differentiate between code we actually want to change and any false positives which might
-  // come up here.)
-  if (Object.keys(DATA_FETCHING_FUNCTIONS).every(functionName => !userCode.includes(functionName))) {
-    return userCode;
-  }
+  // We know one or the other will be defined, depending on the version of webpack being used
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { projectDir } = 'getOptions' in this ? this.getOptions() : this.query;
 
-  const templatePath = path.resolve(__dirname, '../templates/dataFetchersLoaderTemplate.js');
-  // make sure the template is included when runing `webpack watch`
-  this.addDependency(templatePath);
+  // Proxy the processed file
+  if (!this.resourceQuery.includes('sentry-proxy-loader')) {
```

Comment:
Is this to avoid proxying an already proxied thing? If so, can we say so here, and mention that we add the that flag in the proxy module we create below?

Reply:
--> 4c3c3cecd4dd6101a1bfa912180e865ae5e54b4c